### DISCUSSION TITLE
[6.7.2] rebooting a worker

### DISCUSSION THREAD

**Discussion by username-is-already-taken2** [concourse/concourse#6570] [Help & Support]
Hi there, 

Could I get a steer for how other people manage the process of rebooting their workers?  Everything normally runs great except when I need to patch the underling os and reboot a worker.  At the moment I land the worker, stop the service and reboot.

My worker is coming back with the same name, volumes etc but after the reboot the process will start but as is the case today, none of the check tasks work, they are all erroring with the message "guardian.run.exec.exec-with-bndl.create-workdir-failed" that had an exit status of 2.

Should I be clearing out the workdir after each reboot, do I need to prune the worker before restarting again?

Appreciate any advice.

Best Regards

Gary

**Comment by taylorsilva** [comment]
> clearing out the workdir after each reboot

The helm chart does this: https://github.com/concourse/concourse-chart/blob/f06f31d699529e211834887d95a33fb85abfc675/templates/worker-statefulset.yaml#L73
The `workdir` in the chart is on a persistent volume, so it there's a chance there's old stuff in it if the container is recreated.

Probably makes sense for you to do the same. Not sure why it's necessary, I just remembered that it's done in the helm chart.

**Comment by piroux** [comment]
_On docker-compose here (k8s was not an option unfortunately)._

I do not persist the worker state directory between restarts of workers.
Before "worker start", I explicitly trash the volumes generated by the worker as an init step before starting the worker.
Before "worker stop", I do land the worker.

I have to admit that I do not check that the worker-landing is complete before stopping it. Indeed the CLI command `land-worker` is async. (I think I might be interested in a `--wait` flag by the way.)

**Reply by piroux** [reply]
Furthermore, here is what the documentation has to say about [the life of the worker state](https://concourse-ci.org/concourse-worker.html#worker-resource-utilization):
> All state on disk must not outlive the worker itself; it is all ephemeral. If the worker is re-created (i.e. fresh VM/container and all processes were killed), it should be brought back with an empty disk.
